hibernate-core/src/main/java/org/hibernate/internal/util/collections/CollectionHelper.java

```diff
@@ -361,4 +362,27 @@ public static Properties asProperties(Map<?,?> map) {
 		}
 	}
 
+	@SuppressWarnings( "unchecked" )
+	public static <K,V> void collectMapEntries(BiConsumer<K, V> mapEntryConsumer, Object[] mappings) {
+		// even numbered
+		assert mappings.length %2 == 0;
+
+		for ( int i = 0; i < mappings.length; i += 2 ) {
+			mapEntryConsumer.accept( (K) mappings[i], (V) mappings[i+1] );
+		}
+	}
+
+	@SuppressWarnings( "unchecked" )
+	public static <K,S> Map<K, S> asMap(Object[] elements) {
+		assert elements != null;
+		assert elements.length % 2 == 0;
+
+		final HashMap<K,S> map = new HashMap<>();
+		collectMapEntries( map::put, elements );
+		for ( int i = 0; i < elements.length; i += 2 ) {
+			map.put( (K) elements[ i ], (S) elements[ i+1 ] );
+		}
```

Comment:
Do we put the elements twice? Once is in `collectMapEntries( map::put, elements )` and then we insert them again afterwards?